cmd/controller/app/options/options.go

```diff
@@ -73,6 +73,8 @@ type ControllerOptions struct {
 	EnableCertificateOwnerRef bool
 
 	MaxConcurrentChallenges int
+
+	DepreciatedSetTruncatedLabelOrderSecret bool
```

Comment:
If we go the route of using annotations to handle all internal logic, can we remove this option altogether and instead just always truncate the *label value* to 63 characters (so it always fits), forget about hashing or deduping (we will use annotations not labels from now on anyway) and then this should be resolved?

We can then add a release note to say that we no longer set the label value, and users should instead rely on annotations.

WRT migration path for users coming from older versions - so long as cert-manager does not leave 'orphaned' Order or Secret resources around, I think we're all good 😄. So whatever mechanism we use to *find* relevant Order resources needs to rely on *owner references* (as we've *always* set owner references on Order resources that we create, so it should work in all versions).